lib/iris/fileformats/netcdf/loader.py

```diff
@@ -624,6 +624,34 @@ def inner(cf_datavar):
                             break
                 return match
 
+            result = inner
+    elif len(constraints) > 1:
+        if all(
+            isinstance(constraint, iris._constraints.NameConstraint)
+            and constraint.STASH == "none"
+            for constraint in constraints
+        ):
+
+            def inner(cf_datavar):
+                match = False
```

Comment:
I found the use of "match" rather confusing.  So here's a bunch of proposals which I think make it easier to read.
See what you think ...
```suggestion
                match_any_constraint = False
```

(I've checked that this works !)